packages/core/src/modules/basic-messages/BasicMessageEvents.ts

```diff
@@ -2,14 +2,21 @@ import type { BaseEvent } from '../../agent/Events'
 import type { BasicMessage } from './messages'
 import type { BasicMessageRecord } from './repository'
 
-export enum BasicMessageEventTypes {
+export declare enum BasicMessageEventTypes {
   BasicMessageReceived = 'BasicMessageReceived',
+  BasicMessageSent = 'BasicMessageSent',
 }
-
 export interface BasicMessageReceivedEvent extends BaseEvent {
   type: typeof BasicMessageEventTypes.BasicMessageReceived
   payload: {
     message: BasicMessage
     basicMessageRecord: BasicMessageRecord
   }
 }
+export interface BasicMessageSentEvent extends BaseEvent {
+  type: typeof BasicMessageEventTypes.BasicMessageSent
+  payload: {
+    message: BasicMessage
+    basicMessageRecord: BasicMessageRecord
+  }
+}
```

Comment:
Yeah we can do this if we want to! My biggest thought process was if the end user wanted to listen to just inbound messages, for instance, then with one event they'd have to filter based off of the role internally within the record, which is an annoying amount of overhead. 
We could potentially specify something in the event as a compromise between the two.
So, I am definitely open to adjusting here, I just don't want to close off a potential nice avenue/mechanism for end-developers.

Reply:
My preference would be to make it consistent between record types (so we create a low level event type). The for more specific use cases we can always create easy to use utility methods.

Thinking from the other side, if you want to listen to both event types, having them separated also adds overhead (cause you have to merge the two event streams)

What ACA-Py has is a `wait_for_event` method that also takes a condition. If we also use that approach users will be able to reuse filters. And we could even export them (or make it an extension library)


```
agent.events.on(BasicMessageEventTypes.StateChanged, listener, basicMessageReceived)
```

We've had discussions to even make the current events more general, as it would allow even more generalized event handling. E.g. a stateChanged event could apply for all records that have state. Currently you have to add a separate listener for each record type.

Reply:
Yeah making some better event handling capabilities makes a lot of sense to me. We'll go ahead and adjust to be consistent with AFJ as it is now and improve the events as a whole in the future 👍